test/FastTests/Corax/StreamingOptimization.cs

```diff
@@ -0,0 +1,224 @@
+            .GetIndexQuery());
+    
+    [Fact] //MultiTermMatch => cannot optimize
+    public async Task SortingMatchIsNotSkippedOnMultiTermMatch() => await TestQueryBuilder<SortingMatch>(session => 
+        session.Advanced.AsyncDocumentQuery<Dto, DtoIndex>()
+            .WhereLessThan(p => p.First, 10)
+            .OrderBy(x => x.First)
+            .GetIndexQuery());
+    
+    [Fact] // where Name = X and Field < 1 order by Name => where Name = x and Field < 1
+    public async Task SortingMatchIsSkippedWhenIsAndBinaryMatch() => await TestQueryBuilder<BinaryMatch>(session => 
+        session.Advanced.AsyncDocumentQuery<Dto, DtoIndex>()
+            .WhereEquals(p => p.Name, "maciej")
+            .AndAlso()
+            .WhereLessThan(p => p.First, 10)
+            .OrderBy(x => x.Name)
+            .GetIndexQuery());
+    
+    [Fact] // where Name = X or Field < 1 order by Name
+    public async Task SortingMatchIsNotSkippedWhenIsOrBinaryMatch() => await TestQueryBuilder<SortingMatch>(session => 
+        session.Advanced.AsyncDocumentQuery<Dto, DtoIndex>()
+            .WhereEquals(p => p.Name, "maciej")
+            .OrElse()
+            .WhereLessThan(p => p.First, 10)
+            .OrderBy(x => x.Name)
+            .GetIndexQuery());
+    
+    [Fact] // where (Name = x and F < 1) and (S = 2 and F < 2 ) order by Name => skip order by
+    public async Task BinaryMatchOfBinaryMatchAnd() => await TestQueryBuilder<BinaryMatch>(session => 
+        session.Advanced.AsyncDocumentQuery<Dto, DtoIndex>()
+                .OpenSubclause()
+                    .WhereEquals(p => p.Name, "maciej")
+                        .AndAlso()
+                    .WhereLessThan(p => p.First, 10)
+                .CloseSubclause()
+            .AndAlso()
+                .OpenSubclause()
+                    .WhereEquals(p => p.Second, 2)
+                        .AndAlso()
+                    .WhereLessThan(p => p.First, 10)
+                .CloseSubclause()
+            .OrderBy(x => x.Name)
+            .GetIndexQuery());
+    
+    [Fact] // where (Name = x and F < 1) and (S = 2 and F < 2 ) order by Name => skip order by
+    public async Task BinaryMatchOfBinaryMatchOr() => await TestQueryBuilder<SortingMatch>(session => 
+        session.Advanced.AsyncDocumentQuery<Dto, DtoIndex>()
+                .OpenSubclause()
+                    .WhereEquals(p => p.Name, "maciej")
+                        .AndAlso()
+                    .WhereLessThan(p => p.First, 10)
+                .CloseSubclause()
+            .OrElse()
+                .OpenSubclause()
+                    .WhereEquals(p => p.Second, 2)
+                        .AndAlso()
+                    .WhereLessThan(p => p.First, 10)
+                .CloseSubclause()
+            .OrderBy(x => x.Name)
+            .GetIndexQuery());
+    
+    [Fact] // where (Name = x or F < 1) and (S = 2 and F < 2 ) order by Name => cant skip order by
```

Comment:
probably need a test where we are querying by name as the second argument with an and
We can change the order of arguments:

```
// where (F < 1 and Name = x) and (S = 2 and F < 2 ) order by Name => should skip order by
```